R/read_h5ad.R

```diff
@@ -53,7 +53,9 @@ read_h5ad <- function(
   )
 
   if (!is.null(fun)) {
-    fun(adata, ...)
+    res <- fun(adata, ...)
+    adata$close()
+    res
```

Comment:
Now that the implementation of close is fixed, I think this is no longer necessary.

```suggestion
    fun(adata, ...)
```